experimental/grpc_attestation/src/attestation.rs

```diff
@@ -48,15 +48,19 @@ impl Receiver {
     }
 }
 
-struct RequestHandler {
+struct RequestHandler<F> {
     /// Utility object for decrypting and encrypting messages using a Diffie-Hellman session key.
     encryptor: AeadEncryptor,
     /// Handler function that processes data from client requests and creates responses.
-    handler: fn(&[u8]) -> Vec<u8>,
+    handler: F,
 }
 
-impl RequestHandler {
-    pub fn new(encryptor: AeadEncryptor, handler: fn(&[u8]) -> Vec<u8>) -> Self {
+impl<F, S> RequestHandler<F>
+where
+    F: Send + Sync + Clone + FnOnce(Vec<u8>) -> S,
```

Comment:
Would it make sense to make this a Fn (not FnOnce) and not Clone?

Reply:
It won't work. `Fn` means the closure must be `async` without `move`. Currently, such async functions cannot have parameters. But we need the closure to also take a parameter.